src/tests/Microsoft.PowerFx.Interpreter.Tests/IRTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Linq;
+using System.Threading;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    public class IRTests
+    {
+        [Fact]
+        public void ValidateNoRecordToRecordAggregateCoercion()
+        {
+            var tableType = TableType.Empty().Add(new NamedFormulaType(new TypedName(DType.Currency, new DName("Currency"))));
+
+            var symbols = new SymbolTable();
+            symbols.EnableMutationFunctions();
+            symbols.AddVariable("MyTable", tableType);
+
+            var engine = new RecalcEngine(new PowerFxConfig());
+            var checkResult = engine.Check("Patch(MyTable, { Currency: 1.2 }, { Currency: 1.5 })", new ParserOptions() { AllowsSideEffects = true }, symbolTable: symbols);
+
+            checkResult.ThrowOnErrors();
+
+            var runtimeConfig = new SymbolValues();
+            runtimeConfig.Add("MyTable", TableValue.NewTable(tableType.ToRecord()));
+            
+            var evalResult = checkResult.GetEvaluator().EvalAsync(CancellationToken.None, runtimeConfig).Result;
```

Comment:
This isn't really testing the IR, it's testing the evaluation. Should probably also walk the IR and assert that a AggregateCoercionNode isn't present, right?

Reply:
fixed